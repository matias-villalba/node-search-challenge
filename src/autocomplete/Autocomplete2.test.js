const Autocomplete2 = require('./Autocomplete2');
const Suggestions = require('./Suggestions');

describe('Autocomplete2', () => {
  it('should perform a case sensitive prefix search', () => {
    const data = Suggestions.load();
    const autocomplete = new Autocomplete2(data);

    const results = autocomplete.performSearch('Biology');

    expect(results.length).toEqual(6);
    expect(results[0]).toMatchSnapshot();
    expect(results[4]).toMatchSnapshot();
  });
});
