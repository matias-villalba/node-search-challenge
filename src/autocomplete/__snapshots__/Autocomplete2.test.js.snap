// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Autocomplete2 should perform a case sensitive prefix search 1`] = `
Object {
  "attribution": "Raven et al",
  "briefDescription": "",
  "chapterPrice": 3.99,
  "copyrightYear": 2010,
  "description": "Need Raven Biology for your class? Try the Inkling version! Breeze through homework with interactive quizzes, visual guided tours, and quick reference tools like an interactive glossary. Itâ€™s everything you need for your course, evolved. No book required.",
  "edition": "9th edition",
  "price": 139.99,
  "primaryCategory": "Science",
  "publisher": "McGraw-Hill",
  "subCategories": Array [],
  "subtitle": "",
  "thumbnail": "data/thumbnails/book-13.jpeg",
  "title": "Biology",
  "url": "https://www.inkling.com/store/book/biology-peter-raven-9th/",
}
`;

exports[`Autocomplete2 should perform a case sensitive prefix search 2`] = `
Object {
  "attribution": "Rene Fester Kratz and Donna Rae Siegfried",
  "briefDescription": "",
  "chapterPrice": 1.99,
  "copyrightYear": 2010,
  "description": "Want to see what you're made of? Get Biology For Dummies on Inkling. Whether on iPad, iPhone or the web, easily navigate your book with hundreds of live cross-links with the book and to the web. Dive beneath the surface with Test Yourself figures, Guided Tours, and Slideshows. There's even a powerful search engine for finding just what you need. Grasping the complexity of life on earth just got easier.",
  "edition": "2nd edition",
  "price": 14.99,
  "primaryCategory": "Science",
  "publisher": "John Wiley & Sons, Inc",
  "subCategories": Array [],
  "subtitle": "",
  "thumbnail": "data/thumbnails/book-10.jpeg",
  "title": "Biology For Dummies",
  "url": "https://www.inkling.com/store/book/biology-dummies-kratz-siegfried-2nd/",
}
`;

exports[`Autocomplete2 should return results that match with prefix that has several words 1`] = `
Object {
  "attribution": "Rene Fester Kratz and Donna Rae Siegfried",
  "briefDescription": "",
  "chapterPrice": 1.99,
  "copyrightYear": 2010,
  "description": "Want to see what you're made of? Get Biology For Dummies on Inkling. Whether on iPad, iPhone or the web, easily navigate your book with hundreds of live cross-links with the book and to the web. Dive beneath the surface with Test Yourself figures, Guided Tours, and Slideshows. There's even a powerful search engine for finding just what you need. Grasping the complexity of life on earth just got easier.",
  "edition": "2nd edition",
  "price": 14.99,
  "primaryCategory": "Science",
  "publisher": "John Wiley & Sons, Inc",
  "subCategories": Array [],
  "subtitle": "",
  "thumbnail": "data/thumbnails/book-10.jpeg",
  "title": "Biology For Dummies",
  "url": "https://www.inkling.com/store/book/biology-dummies-kratz-siegfried-2nd/",
}
`;
